projects/rocprofiler-systems/source/bin/rocprof-sys-avail/common.cpp

```diff
@@ -307,30 +308,60 @@ process_categories(parser_t& p, const str_set_t& _category_options)
 {
     category_view = p.get<str_set_t>("categories");
     std::vector<std::function<void()>> _shorthand_patches{};
+
+    // Helper to do case-insensitive string comparison
+    auto _tolower = [](const std::string& in) {
+        std::string out = in;
+        std::transform(out.begin(), out.end(), out.begin(),
+                       [](unsigned char c) { return std::tolower(c); });
+        return out;
+    };
+
+    // Helper to find case-insensitive match in category options
+    auto _find_category = [&_category_options,
```

Comment:
You could optimize this with caching somewhere the lowercase categories. Since now `_find_category ` lambda is creates a new string copy for every comparison. Some quick math, for around 30 categories you would have made more than 500 string copies. Utilizing `unordered_map` could be some simple solution, like this:

```suggestion
    std::unordered_map<std::string, std::string> _category_map;
    for(const auto& opt : _category_options)
    {
        _category_map[_tolower(opt)] = opt;
    }
```

And then the `_find_category` could look like this:
```suggestion
    // Helper to find case-insensitive match in category options
    auto _find_category = [&_category_map,
                           &_tolower](const std::string& input) -> std::string {
        auto input_lower = _tolower(input);
        auto it = _category_map.find(input_lower);
        return (it != _category_map.end()) ? it->second : "";
    };
```

Reply:
Agree. Made the changes.